scripts/publish-pvkeys.sh

```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+set -eo pipefail
+
+KEYDIR="/tmp/coda_cache_dir/"
+
+# Only publish if we have AWS Creds
+if [ -z "$AWS_ACCESS_KEY_ID" ]; then
+    echo "WARNING: AWS_ACCESS_KEY_ID not set, publish commands not run"
+else
+    # Only publish if there are generated files in place.
+    shopt -s nullglob dotglob
```

Comment:
What does `shopt` do in this context?

Reply:
Was using a bash pattern for detecting if ANY files exist in a dir..

nullglob
If set, Bash allows filename patterns which match no files to expand to a null string, rather than themselves.

dotglob
If set, Bash includes filenames beginning with a ‘.’ in the results of filename expansion. The filenames ‘.’ and ‘..’ must always be matched explicitly, even if dotglob is set.

I think we might need nullglob if it's empty...

I think we might not need dotglob, but it doesn't hurt.

Reply:
Sweet